### DISCUSSION TITLE
[C#]请问能否脱离UnityEditor进行代码生成？

### DISCUSSION THREAD

**Discussion by minim271** [Tencent/puerts#1511] [Q&A] [answered]
我目前将PuerTS用于Godot4(C#)，可以理解为常规的dotnet项目，已经走通了JS侧反射调用CS侧，希望能够通过代码生成避开反射开销和生成.d.ts。

目前C#这边代码生成好像只适配了UnityEditor的菜单，如果没理解错代码生成功能好像并不是和Unity耦合的，应该可以实现在单独CLI工程中执行代码生成（根据TypeList生成代码和.d.ts至特定目录）。

我尝试了一下使用PuerTS的C#源码在一个单独的CLI工程中生成代码，但没能理解该如何调用API（编译通了，但调用方式应该不对，并没输出文件）。

如果可能希望能够有一个简单的示例，对于非Unity项目来说应该都会需要。

**Comment by zombieyang** [comment, marked as answer]
https://github.com/Tencent/puerts/blob/master/unity/test/dotnet/Src/WrapperGenerator/WrapperGenerate.cs
这里有一个在dotnet下生成wrapper的示例。我的单元测试本就有在纯dotnet下跑
